typescript/src/lib/utils/ledger.ts

```diff
@@ -0,0 +1,177 @@
+    if (!this._account || !this._account.address) {
+      throw new AccountNotFoundError()
+    }
+    this._windowMessageTransport.connect()
+    const buffer = await this._walletApiClient.message.sign(
+      this._account.id,
+      Buffer.from(message)
+    )
+    this._windowMessageTransport.disconnect()
+    return buffer.toString()
+  }
+
+  async signTransaction(
+    transaction: ethers.providers.TransactionRequest
+  ): Promise<string> {
+    if (!this._account || !this._account.address) {
+      throw new AccountNotFoundError()
+    }
+
+    const { value, to, nonce, data, gasPrice, gasLimit } = transaction
+
+    const ethereumTransaction: any = {
+      family: "ethereum" as const,
+      amount: value ? new BigNumber(value.toString()) : new BigNumber(0),
+      recipient: to ? to : AddressZero,
+    }
+
+    if (nonce) ethereumTransaction.nonce = nonce
+    if (data)
+      ethereumTransaction.data = Buffer.from(
+        Hex.from(data.toString()).toString(),
+        "hex"
+      )
+    if (gasPrice)
+      ethereumTransaction.gasPrice = new BigNumber(gasPrice.toString())
+    if (gasLimit)
+      ethereumTransaction.gasLimit = new BigNumber(gasLimit.toString())
+
+    this._windowMessageTransport.connect()
+    const buffer = await this._walletApiClient.transaction.sign(
+      this._account.id,
+      ethereumTransaction
+    )
+    this._windowMessageTransport.disconnect()
+    return buffer.toString()
+  }
+
+  async sendTransaction(
+    transaction: Deferrable<ethers.providers.TransactionRequest>
+  ): Promise<ethers.providers.TransactionResponse> {
+    if (!this._account || !this._account.address) {
+      throw new AccountNotFoundError()
+    }
+
+    const { value, to, nonce, data, gasPrice, gasLimit } = transaction
+
+    const ethereumTransaction: any = {
+      family: "ethereum" as const,
+      amount: value ? new BigNumber(value.toString()) : new BigNumber(0),
+      recipient: to ? to : AddressZero,
+    }
+
+    if (nonce) ethereumTransaction.nonce = nonce
+    if (data)
+      ethereumTransaction.data = Buffer.from(
+        Hex.from(data.toString()).toString(),
+        "hex"
+      )
+    if (gasPrice)
+      ethereumTransaction.gasPrice = new BigNumber(gasPrice.toString())
+    if (gasLimit)
+      ethereumTransaction.gasLimit = new BigNumber(gasLimit.toString())
+
+    this._windowMessageTransport.connect()
+    const transactionHash =
+      await this._walletApiClient.transaction.signAndBroadcast(
+        this._account.id,
+        ethereumTransaction
+      )
+    this._windowMessageTransport.disconnect()
+
+    const transactionResponse = await this.provider?.getTransaction(
+      transactionHash
+    )
+
+    if (!transactionResponse) {
+      throw new Error("Transaction response not found!")
+    }
+
+    return transactionResponse
+  }
+
+  connect(provider: ethers.providers.Provider): Signer {
+    return new LedgerLiveAppEthereumSigner(provider)
+  }
```

Comment:
> Is returning a new instance a typical pattern used within Ethers lib? Given the provider is readonly, I suppose that's the case but just want to confirm.

Yes, the method is required when implementing `Signer` from `ethers` lib. I'm not sure if and where exactly it is used in our flow though 🤔 

> Also, shouldn't we pass `_windowMessageTransport` and `_walletApiClient` to the new instance? Without that, the new instance is created with the defaults, which may be against intuition. (...)

Good point, but it is not needed to pass them anymore, because i've removed them from the constructor. See https://github.com/keep-network/tbtc-v2/pull/743#discussion_r1392431765 for more details.

> (...) When I call `connect`, I don't expect anything beside the provider changes in the signer state. I also think we should pass the account and set it through `setAccount`. WDYT?

I agree. Fixed in 848142fe21624a19c8af111aa09b004d52362f44 where i set the current account for the new instance. Can you double check if that's what you meant?